ngraph/frontend/onnx/frontend/src/op/if.cpp

```diff
@@ -0,0 +1,59 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "op/if.hpp"
+
+#include "core/graph.hpp"
+#include "ngraph/node.hpp"
+#include "ngraph/opsets/opset8.hpp"
+
+namespace ngraph {
+namespace onnx_import {
+namespace op {
+namespace set_1 {
+OutputVector if_op(const Node& node) {
+    const auto& ng_inputs = node.get_ng_inputs();
+
+    const auto& subgraphs = node.get_subgraphs();
+    NGRAPH_CHECK(subgraphs.count("then_branch") == 1, "Missing 'then_branch' attribute");
+    auto then_subgraph = subgraphs.at("then_branch");
+    const auto& then_params = then_subgraph->get_ng_parameters();
+    auto then_branch =
+        std::make_shared<Function>(then_subgraph->get_ng_outputs(), then_params, then_subgraph->get_name());
+    NGRAPH_CHECK(subgraphs.count("else_branch") == 1, "Missing 'else_branch' attribute");
+    auto else_subgraph = subgraphs.at("else_branch");
+    const auto& else_params = else_subgraph->get_ng_parameters();
+    auto else_branch =
+        std::make_shared<Function>(else_subgraph->get_ng_outputs(), else_params, else_subgraph->get_name());
+
+    auto if_node = std::make_shared<ngraph::opset8::If>(ng_inputs.at(0));
+    if_node->set_then_body(then_branch);
+    if_node->set_else_body(else_branch);
+    const auto then_branch_inputs_from_parent = then_subgraph->get_inputs_from_parent();
+    auto then_param = then_params.cbegin();
+    for (const auto& from_parent : then_branch_inputs_from_parent) {
+        if_node->set_input(from_parent, *then_param, nullptr);
+        then_param++;
```

Comment:
Is there any chance that the number of `then_params` and `then_branch_inputs_from_parent` will be different? If so, I suggest adding a check before this loop. Otherwise you might attempt to increment an iterator that has already reached `then_params.end()`.

Reply:
probably in a buggy model. I added a check for that